app/src/common/chromium/com/igalia/wolvic/browser/api/impl/TabWebContentsObserver.java

```diff
@@ -0,0 +1,80 @@
+package com.igalia.wolvic.browser.api.impl;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.igalia.wolvic.browser.api.WSession;
+
+import org.chromium.content_public.browser.LifecycleState;
+import org.chromium.content_public.browser.NavigationHandle;
+import org.chromium.content_public.browser.WebContents;
+import org.chromium.content_public.browser.WebContentsObserver;
+import org.chromium.url.GURL;
+
+public class TabWebContentsObserver extends WebContentsObserver {
+    private @NonNull WebContents mWebContents;
```

Comment:
I don't think you need this extra reference. mWebContents from WebContentsObserver is already set in constructor, and it is protected.

Reply:
Done!